packages/wifi_basic/lib/wifi_info.dart

```diff
@@ -0,0 +1,62 @@
+part of 'wifi_basic.dart';
+
+class WiFiInfo {
+  static const Equality _equality = DeepCollectionEquality();
+
+  final String ssid;
+  final String bssid;
+  final WiFiNetworkSecurity security;
+  final bool isHidden;
+  final int rssi;
+  final double signalStrength;
+  final bool hasInternet;
+  final WiFiGenerations generation;
+
+  bool get isNull => ssid.isEmpty;
+
+  @visibleForTesting
+  WiFiInfo.fromMap(Map map)
+      : ssid = map["ssid"] as String,
+        bssid = map["bssid"] as String,
+        security =
+            WiFiNetworkSecurityExtension.fromInt(map["security"] as int?),
+        isHidden = map["isHidden"] as bool,
+        rssi = map["rssi"] as int,
+        signalStrength = map["signalStrength"] as double,
+        hasInternet = map["hasInternet"] as bool,
+        generation =
+            WiFiGenerationsExtension.fromInt(map["generation"] as int?);
+
+  @override
```

Comment:
Now that I think of `==` override and `hashCode` looks unnecessary - what could be the use case to do it? If it has solid use case then we might want to implement it for all classes.

Reply:
It also need `collection` dependency.

Reply:
I see the WifiInfo as data class, since it does exatcly it: holds data. And every data class should override `==` and `hashCode`, otherwise, two identical classes would be considered different, which is wrong.

Reply:
The collection dependency isn't exatcly required. It's totally possible override those methods without it, but this packages already handle lists, sets, deep objects andsome more use cases, that otherwise, we would have to handle by ouselves.

And since the collections package is maintained by the dart team, I don't see a problem using it.

Reply:
A concrete use case can be seen in tests, where I compare `info` against the object `WifiInfo`. That comparison would have failed without those overrides